kafka-client/src/main/java/dev/responsive/kafka/internal/db/FlushManager.java

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ *  * Copyright 2023 Responsive Computing, Inc.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package dev.responsive.kafka.internal.db;
+
+import dev.responsive.kafka.internal.db.partitioning.TablePartitioner;
+import dev.responsive.kafka.internal.stores.RemoteWriteResult;
+
+public interface FlushManager<K, P> {
+
+  TablePartitioner<K, P> partitioner();
+
+  RemoteWriter<K, P> createWriter(final P tablePartition);
+
+  RemoteWriteResult<P> preFlush();
+
+  RemoteWriteResult<P> postFlush(final long consumedOffset);
+
+  String logPrefix();
+
+  default String failedFlushMsg(final long batchOffset) {
```

Comment:
can we include these in the return values instead of putting methods in the interface?